core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/PauseManager.scala

```diff
@@ -34,15 +43,34 @@ class PauseManager(controlOutputPort: ControlOutputPort) {
       if(level >= pausePrivilegeLevel.get())
         pausePrivilegeLevel.set(level)
      */
-    pausePrivilegeLevel.getAndUpdate(i => if (Paused >= i) Paused else i)
+    pausePrivilegeLevel.getAndUpdate { i =>
+      if (Paused >= i) {
+        Paused
+      } else i
+    }
+
   }
 
+  def registerPromise(workflowPromise: WorkflowPromise[ExecutionPaused]): Unit = {
+    if (isPaused) {
+      controlOutputPort.sendTo(
+        VirtualIdentity.Self,
+        ReturnPayload(workflowPromise.ctx, ExecutionPaused())
+      )
+    } else {
+      promiseFromActorThread = workflowPromise
+    }
+  }
+
+  def isPaused: Boolean = isPauseSet() && dpThreadBlocker != null
```

Comment:
This check is not true. After our bug fix, dpThread won't be null anytime (after the system is paused once). And isPauseSet is just checking the flags.